project-base/storefront/components/Pages/Order/TransportAndPayment/helpers.ts

```diff
@@ -0,0 +1,356 @@
+    changeTransportHandler: ChangeTransportHandler,
+    changePaymentHandler: ChangePaymentHandler,
+) => {
+    const clearPacketeryPickupPoint = usePersistStore((store) => store.clearPacketeryPickupPoint);
+
+    const resetTransportAndPayment = async () => {
+        await changeTransportHandler(null, null);
+        await changePaymentHandler(null, null);
+        clearPacketeryPickupPoint();
+    };
+
+    return resetTransportAndPayment;
+};
+
+export const getLastOrderPickupPlace = (
+    lastOrder: LastOrderFragmentApi,
+    lastOrderPickupPlaceIdentifier: string,
+    lastOrderPickupPlaceFromApi: ListedStoreFragmentApi | undefined | null,
+    packeteryPickupPoint: ListedStoreFragmentApi | null,
+): ListedStoreFragmentApi | null => {
+    if (packeteryPickupPoint?.identifier === lastOrderPickupPlaceIdentifier) {
+        return packeteryPickupPoint;
+    }
+
+    if (lastOrderPickupPlaceFromApi?.identifier) {
+        return getGtmPickupPlaceFromStore(lastOrderPickupPlaceFromApi);
+    }
+
+    return getGtmPickupPlaceFromLastOrder(lastOrderPickupPlaceIdentifier, lastOrder);
+};
+
+type TransportAndPaymentErrorsType = {
+    transport: {
+        name: 'transport';
+        label: string;
+        errorMessage: string | undefined;
+    };
+    payment: {
+        name: 'payment';
+        label: string;
+        errorMessage: string | undefined;
+    };
+    goPaySwift: {
+        name: 'goPaySwift';
+        label: string;
+        errorMessage: string | undefined;
+    };
+};
+
+export const getTransportAndPaymentValidationMessages = (
+    transport: Maybe<TransportWithAvailablePaymentsAndStoresFragmentApi>,
+    pickupPlace: Maybe<ListedStoreFragmentApi>,
+    payment: Maybe<SimplePaymentFragmentApi>,
+    paymentGoPayBankSwift: Maybe<string>,
+    t: Translate,
+) => {
+    const errors: Partial<TransportAndPaymentErrorsType> = {};
+
+    if (!transport) {
+        errors.transport = {
+            name: 'transport',
+            label: t('Choose transport'),
+            errorMessage: t('Please select transport'),
+        };
+    } else {
+        if (transport.isPersonalPickup && !pickupPlace?.identifier) {
+            errors.transport = {
+                name: 'transport',
+                label: t('Choose transport'),
+                errorMessage: t('Please select transport with a personal pickup place'),
+            };
+        }
+        if (!payment) {
+            errors.payment = {
+                name: 'payment',
+                label: t('Choose payment'),
+                errorMessage: t('Please select payment'),
+            };
+        }
+        if (payment?.goPayPaymentMethod?.identifier === 'BANK_ACCOUNT' && !paymentGoPayBankSwift) {
+            errors.goPaySwift = {
+                name: 'goPaySwift',
+                label: t('Choose your bank'),
+                errorMessage: t('Please select your bank'),
+            };
+        }
+    }
+
+    return errors;
```

Comment:
Maybe better readable? I will leave it on you.

```
    const errors: Partial<TransportAndPaymentErrorsType> = {};

    if (!transport) {
        errors.transport = {
            name: 'transport',
            label: t('Choose transport'),
            errorMessage: t('Please select transport'),
        };
        
        return errors;
    } 
    
        if (transport.isPersonalPickup && !pickupPlace?.identifier) {
            errors.transport = {
                name: 'transport',
                label: t('Choose transport'),
                errorMessage: t('Please select transport with a personal pickup place'),
            };
        }
        if (!payment) {
            errors.payment = {
                name: 'payment',
                label: t('Choose payment'),
                errorMessage: t('Please select payment'),
            };
        }
        if (payment?.goPayPaymentMethod?.identifier === 'BANK_ACCOUNT' && !paymentGoPayBankSwift) {
            errors.goPaySwift = {
                name: 'goPaySwift',
                label: t('Choose your bank'),
                errorMessage: t('Please select your bank'),
            };
        }
   

    return errors;
```

Reply:
Yeah, could be. Changed.